### PR TITLE
Adding solver settings & shadow price

### PR DESCRIPTION
**Pull request by Saeid3**
In this PR, I added shadow prices to the output of LP solutions. I also added an object of solver settings to the API with values of mipGap, tmLim & msgLev which are the important ones.
I pass these settings as function args in glpk.js.c hardcoded. A better way would be to create an object of these settings in js and pass it to the solve_mip (something like following), but I couldn't manage to do it.
```C
int solve_mip(glp_prob *P, setting_obj settings) {
	glp_iocp parm;
	glp_init_iocp(&parm);
	parm.presolve = 1;
	parm.msg_lev = msg_lev;
	parm.mip_gap = settings.mip_gap;
	parm.tm_lim = settings.tm_lim;
	return glp_intopt(P, &parm);
}
```
This PR addresses https://github.com/jvail/glpk.js/issues/16

### CODE REVIEW

**Comment by jvail** [comment]
First of all thank you for this contribution - very interesting improvements. Just two comments here the rest will take a bit of time:

- yes, passing a struct/obj to the c solve functions would be better. Its doable but I thing for now (since there are only a few settings  right now it is not a big deal)
- should I create an AUTHORS file and add you and probably your colleague as well?

I will add comments and suggestions to the sources - quite a lot I am afraid. I can of course do some of them as well. But that might take some time.

**Comment by Saeid3** [comment]
Thank you for the suggestion.
- about the AUTHORS that sounds good to us

Also please go ahead and leave your comments and suggestions, that would help a lot

**Review by jvail — COMMENTED** [review]
> Thank you for the suggestion.
> 
>     * about the AUTHORS that sounds good to us

Sure, I'll add one. Or if you like you can include one here with your name.

File: glpk.d.ts [hunk 1]
```diff
@@ -26,7 +27,8 @@ interface Result {
     result: {
         status: number;
         z: number;
-        vars: {[key:string]: number}
+        vars: {[key:string]: number};
+        cons: {[key:string]: {dual: number}}
```
[the remark this answers is not in the document]
Comment (Saeid3) [reply glpk.d.ts:31]: I was thinking the constraint has other attributes that maybe we add in future, sth like the following that I took from Gurobi attributes:
image

File: test/test.js [hunk 2]
```diff
@@ -37,6 +39,19 @@ require('../glpk.js').then(glpkjs => {
...
+  
+  tape('The time limit should kill the solver before finding optimal solution', { timeout: 99999 }, t => {
+    const problem = 'mip2'
+    let json = JSON.parse(fs.readFileSync(`${__dirname}/data/${problem}.json`).toString());
+    const settings = _.defaults({tmLim: 1}, json.settings)
+    
+    const sol = glpkjs.solve(json, settings);
 
-});
+    t.equal(1, sol.result.status, 'solution is undefined')
```
[the remark this answers is not in the document]
Comment (Saeid3) [reply test/test.js:50]: For the mipGap, the problem should be hard enough that the solver can't solve the exact solution for a given time. The test cases I have for mip gap are very big problems, will try to find a smaller one

**Comment by jvail** [comment]
Alright, thank you @Saeid3 for the changes. I think I misunderstood what you try do do with "dual". I thought you want to achieve the same as passing "--dual" to glpsol. If we just add an additional output (const dual) that it wont be optional.

I looked a bit through the glpk.pdf and I think I will probably change the naming of the options so they are identical with glpsol.
For a version 4 with a changed API i might need more time since I would like to change a lot on the emscripten part (e.g ES6 export etc.). So I will change the solve function to allow both passing either the msg_lvl or an options object.
Then I can quickly make a 3.2 release including all of your changes.

Could you add an AUTHORS file with your and maybe your colleagues names? - if you want. I don't know what should appear there.

**Comment by Saeid3** [comment]
Thank you @jvail for reviewing the changes. I changed the cons object not to be optional. but dual still remains optional because we won't have dual for problems that aren't LP.
I also didn't know what should a AUTHOR file include, I just made a `.md` file manually with the names. Feel free to change as you like.

**Comment by jvail** [comment]
All right @Saeid3. Thank for the Authors file and your changes. I try to get it done and released as soon as possible.

**Pull request merged**